### DISCUSSION TITLE
Support of devices

### DISCUSSION THREAD

**Discussion by Davedonaldson** [SafeExamBrowser/seb-win-refactoring#169] [Q&A]
Hello,

I am currently looking into the possibility of using SafeExamBrowser. Does this application support external Mouse and Keyboard, as some of my students use a desktop instead of a laptop. 

If so, can we check if there are multiple Bluetooth / USB/ PS2 / MAC devices connected? Its kind of a nightmare to filter everything with our current system - I don't want to exclude students because they use a MAC or an old Desktop, but also want to maintain integrity of our tests.

Cheers

**Comment by dbuechel** [comment]
Yes, SEB allows the usage of external keyboard and mouse. There however isn't any restriction on how many keyboards or mice are connected to the computer. The only configuration possibility is to allow or prohibit certain keys or key combinations: https://safeexambrowser.org/windows/win_usermanual_en.html#HookedKeysPane.

**Comment by danschlet** [comment]
I don't think it makes sense for us to prevent people using external mouses/trackpads etc. Many users, especially with Windows notebooks prefer to use an external mouse (as many PC notebooks have terrible built-in trackpads), many would for sure complain if you wouldn't allow that. 

We might sometimes in future add a logging and SEB Javascript API feature to list what kind/number of input devices are connected, then developers of web assessment solutions could decide themselves how to deal with that.

SEB for macOS also (since long) supports the settings for allowed number of displays and to force using an internal display ([see SEB for Mac manual](https://safeexambrowser.org/macosx/mac_usermanual_en.html#SecurityPane)). In the Mac version you can even configure if Macs without built-in display (Mac Mini, Mac Pro) should be allowed to be used or not.

**Comment by Davedonaldson** [comment]
Hi Damian and Daniel,

Thank you for your extensive feedback, and sorry for not getting to you sooner. Because we fear new lockdowns, we are even thinking about lending special laptops or go with a new plugin. The latter would be a much cheaper option but could be more prone to cheating (?).

Daniel, you said it is not possible to detect multiple screens at the moment- however, I can imagine there are other possibilities to monitor students in another way like: 

**Can you detect multiple input devices?**
**Would it be possible to monitor and match mouse clicks and keystrokes with audio recording?** 

I am not an IT guy, but I thought thjs would make some sense.

Cheers
